clients/java/src/main/java/io/zeebe/client/util/Environment.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright © 2017 camunda services GmbH (info@camunda.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.zeebe.client.util;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Representation of client environment variables
+ *
+ * <p>Useful primarily to allow us to test with custom environment variables
+ */
+public class Environment {
+  // Returns an environment created from the system's environment variables using System#getenv()
+  private static final Environment SYSTEM = new Environment(System.getenv());
+
+  private final Map<String, String> variables;
+
+  public Environment() {
+    this.variables = new HashMap<>();
+  }
+
+  public Environment(final Map<String, String> variables) {
```

Comment:
I think these ctors should then be private and the environment final

Reply:
Should we create a more thread safe singleton ? https://www.geeksforgeeks.org/java-singleton-design-pattern-practices-examples/#targetText=Thread%20Safe%20Singleton%3A%20A%20thread,can't%20access%20it%20simultaneously.&targetText=Pros%3A,Lazy%20initialization%20is%20possible.